include/nil/actor/math/polynomial/polynomial_dfs.hpp

```diff
@@ -348,45 +349,26 @@ namespace nil {
                     val.clear();
                 }
 
-                future<> resize(size_type _sz) {
-                    BOOST_ASSERT_MSG(_sz >= _d, "Can't restore polynomial in the future");
-                    typedef typename value_type::field_type FieldType;
-
-                    value_type omega = crypto3::math::unity_root<FieldType>(this->size());
-
-                    detail::basic_radix2_fft<FieldType>(val, omega.inversed()).get();
-
-                    const value_type sconst = value_type(this->size()).inversed();
-
-                    detail::block_execution(this->size(),
-                                            smp::count,
-                                            [this, sconst](std::size_t begin, std::size_t end) {
-                                                for (std::size_t i = begin; i < end; i++) {
-                                                    val[i] *= sconst;
-                                                }
-                                            })
-                        .get();
-
-                    value_type omega_new = crypto3::math::unity_root<FieldType>(_sz);
-                    val.resize(_sz);
+                void resize(size_type _sz) {
+                    // BOOST_ASSERT_MSG(_sz >= _d, "Can't restore polynomial in the future");
 
-                    detail::basic_radix2_fft<FieldType>(val, omega_new).get();
+                    if (this->size() == 1){
+                        this->val.resize(_sz, this->val[0]);
+                    } else {
+                        typedef typename value_type::field_type FieldType;
 
-                    return make_ready_future<>();
+                        make_evaluation_domain<FieldType>(this->size())->inverse_fft(this->val);
```

Comment:
We didn't call `.get()` here and in some another places. It can be problem, if some operation inside didn't end, but we try to use them result. As the result we get strange behaviour depend on CPU performance. 

Also - write `.get()` and return `future` - best way to show that we use some parallel operations inside.

Reply:
Yes, make_evaluation_domain does not have it, but `inverse_fft` and `fft` return future. 
Okay let's assume we have 
```
void f(...) {
   return detail::block_execution(this->m, smp::count, [&a, &T](std::size_t begin, std::size_t end) {
         for (std::size_t i = begin; i < end; i++) {
             a[i] = a[i] * T[i].inversed();
       }
  });
}

a = [1, 2, 3]
T = [4, 5, 6]
f(...)
print(a[2])
``` 
And we get question
a[2] = 3 or a[2] = 3 * inverse(6)? 
We don't wait execution our future, and we don't know if block already finish all operations and we have changed a, or it's still running.

>Why would the caller care, if it has parallelism or not
In best scenario - he don't care. But now we don't have sheduler, and return future everythere simplest way to understand "can I parallel this function or all my cores already busy?"
For example 
```
void f() {
   for i = 1 ... N
      do smth
}

vod x() {
   for i = 1 ... N
        call f
} 
```
So when we try to improve performance - we look at `f()` and say - wow I want to parallel it. But after we watch at `x()`, and also saw this possibility. But if we already make all cores busy at `f()` - we can't do it.

Reply:
Done.